src/oss/langgraph/sql-agent.mdx

```diff
@@ -101,9 +121,57 @@ Dialect: sqlite
 Available tables: ['Album', 'Artist', 'Customer', 'Employee', 'Genre', 'Invoice', 'InvoiceLine', 'MediaType', 'Playlist', 'PlaylistTrack', 'Track']
 Sample output: [(1, 'AC/DC'), (2, 'Accept'), (3, 'Aerosmith'), (4, 'Alanis Morissette'), (5, 'Alice In Chains')]
 ```
+:::
+:::js
+```typescript
+import fs from "node:fs/promises";
+import path from "node:path";
+
+const url = "https://storage.googleapis.com/benchmarks-artifacts/chinook/Chinook.db";
+const localPath = path.resolve("Chinook.db");
+
+async function resolveDbPath() {
+  const exists = await fs.access(localPath).then(() => true).catch(() => false);
+  if (exists) {
+    console.log(`${localPath} already exists, skipping download.`);
+    return localPath;
+  }
+  const resp = await fetch(url);
+  if (!resp.ok) throw new Error(`Failed to download DB. Status code: ${resp.status}`);
+  const buf = Buffer.from(await resp.arrayBuffer());
+  await fs.writeFile(localPath, buf);
+  console.log(`File downloaded and saved as ${localPath}`);
+  return localPath;
+}
+```
+
+We will use a handy SQL database wrapper available in the `@langchain/classic/sql_db` module to interact with the database. The wrapper provides a simple interface to execute SQL queries and fetch results:
+
+```typescript
+import { SqlDatabase } from "@langchain/classic/sql_db";
```

Comment:
I'm using this wrapper as we already have it. Did we want to modify this to directly use typeorm rather than pulling from classic?

Reply:
Yeah it's doing a bit of work. Mainly around getting db schema in dialect-agnostic way